test/mnlistdiff/data.js

```diff
@@ -0,0 +1,33 @@
+// on dapinet devnet September 2018 (protx 1, 2896)
+const mnListDiff = {
+    baseBlockHash: '3f4a8012763b1d9b985cc77b0c0bca918830b1ef7dd083665bdc592c2cd31cf6',
+    blockHash: '000004543e350b99f43114fe0bf649344a28f4fde6785d80e487d90689ae3918',
```

Comment:
Apologies if I'm wrong, but I think there still is a slight misunderstanding regarding new P2P message getmnlistdiff and the rpc interface which dapi will use to get the equivalent needed to communicate with its connected dapi clients. If I understand correctly this PR implements the parsers for the new P2P messages getmnlistdiff which include the cbTx and is needed for traditional SPV clients to understand these new P2P messages. This is not only needed for legacy SPV clients, but also for DAPI clients that want to check the cbTx's merkleRootMNList field against their own resulting det. MNlist's merkle root. However, the message to test against you added here is the return of the rpc command which only DAPI will use. For testing the P2P mnlistdiff message please use real p2p message.

What a DAPI client (as opposed to a real P2P SPV client) will need is imo:
1. from DAPI 
endpoints which gives him access to both det. masternode list versions
- protx list etc. etc
- protx diff etc. etc.
in this case there is no need to include the whole cbTx, because the field merkleRootMNList which the DAPI client gets from the DAPI endpoint is all that is needed to check if the list it constructs by itself from information it obtained by making request to the DAPI nodes is correct.
2. from dashcore-lib
a function to calculate the merkleroot of the masternodelist
input param would be the whole masternode list the client came up with after applying the changes that came with this new block's deletedMNs and MNlist fields in the rpc call's json result
as described here https://github.com/dashpay/dips/blob/master/dip-0004.md#calculating-the-merkle-root-of-the-masternode-list
If his own calculation results in the same merkle root as the field merkleRootMNList, then it can assume the list to be correct with trust level 0. The DAPI node could have still spoofed him a homemade masternode list at that point. So, if he needs higher trust level then he can check another DAPI node for the merkleRootMNList field at the block height in question.